geoSegment.js

```diff
@@ -258,72 +310,42 @@ async function downloadSessionSeries(firehoseBaseUrl, firehoseApiKey, segments)
  * @param {ScheduledSessionItem[]} items
  */
 async function processScheduledSessionItems(items) {
-  items.forEach(async (scheduledSessionItem) => {
+  for (const scheduledSessionItem of items) {
     // Filter deleted ScheduledSessions
     if (scheduledSessionItem.state === 'deleted') {
       return;
     }
+    // If for some reason, the ScheduledSession doesn't have an ID, don't process it.
+    if (!scheduledSessionItem.id) {
+      return;
+    }
 
     // Filter ScheduledSessions in the past
     if (!scheduledSessionItem.data.startDate || moment(scheduledSessionItem.data.startDate).isBefore(moment())) {
       return;
     }
 
     // Does ScheduledSession have downloaded superEvent?
-    const scheduledSessionSuperEventId = scheduledSessionItem.data.superEvent;
-    const sessionSeriesFilePath = getSessionSeriesFilePath(scheduledSessionSuperEventId);
-    if (!SESSION_SERIES_FILE_PATHS.includes(sessionSeriesFilePath)) {
-      return;
-    }
+    const scheduledSessionSuperEventIdHash = hashString(scheduledSessionItem.data.superEvent);
+    const sessionSeriesFilePath = getSessionSeriesFilePath(scheduledSessionSuperEventIdHash);
 
     // Get SessionSeries
-    const sessionSeries = await fs.readJson(sessionSeriesFilePath);
-
-    // Link ScheduledSession and SessionSeries, and write
-    await linkScheduledSessionAndSessionSeriesAndWrite(scheduledSessionItem.data, sessionSeries);
-  });
-}
-
-async function downloadScheduledSessions(firehoseBaseUrl, firehoseApiKey) {
-  console.log('downloadAndWriteScheduledSessions() - starting');
-  // Get ScheduledSessions from Firehose
-  let nextUrl = `${firehoseBaseUrl}scheduled-sessions`;
-  let backoffTimeInSeconds = RETRY_BACKOFF_MIN;
-  while (true) {
-    let nextNextUrl;
-    let items;
-    try {
-      ({ nextNextUrl, items } = await getFirehosePage(nextUrl, firehoseApiKey));
-    } catch (error) {
-      if (error instanceof Http404Error) {
-        console.error(`ERROR: URL ("${error.url}") not found. Please check the value for <rpde-endpoint>.`);
-        process.exit(1);
-      }
-      const loggableError = (error && error.isAxiosError)
-        ? loggableAxiosError(error)
-        : error;
-      console.warn(`WARN: Retrying [page: "${nextUrl}"] due to error:`, loggableError);
-      if (backoffTimeInSeconds > RETRY_BACKOFF_MAX) {
-        console.error('ERROR: Cannot download Firehose pages, backoff limit reached, terminating script');
-        process.exit(1);
-      }
-      await wait(backoffTimeInSeconds * 1000);
-      backoffTimeInSeconds *= RETRY_BACKOFF_EXPONENTIATION_RATE;
-      continue;
+    if (!(await fs.pathExists(sessionSeriesFilePath))) {
+      return;
+    }
```

Comment:
pathExists check isn't needed if fs.readJson already gives us an existence check (by returning null) (there's a couple other instances of this)